src/main/ApplicationImpl.cpp

```diff
@@ -578,6 +578,18 @@ ApplicationImpl::manualClose()
     return false;
 }
 
+void
+ApplicationImpl::manualClose(std::map<std::string, std::string> const& retMap,
+                             std::string& retStr)
+{
+    if (!mConfig.ENHANCED_MANUAL_CLOSE)
```

Comment:
I ended up doing it the way I did primarily because it means that if we ever decide to change the conditions under which `ENHANCED_MANUAL_CLOSE` is allowed to be set, we'll only have to change those checks, and not have to remember to change anything here.

Paranoia about such things isn't bad, though.  Still, if we _do_ end up changing the conditions, it could be to make them _more_ restrictive, in which case there could be the risk in forgetting to update them here that we later lead someone who notices the inconsistency to decide that the old, more lax requirements are the right ones.  Overall I think I favor leaving it and trusting ourselves to have done the validation correctly when we first set the flag.

Reply:
I'm not sure you understood quite what I meant (not that I necessarily disagree with your decision, I just want to make sure we're speaking the same language). I'm talking about purely internal usage, like
```
Config cfg;
cfg.ENHANCED_MANUAL_CLOSE = true;
...
```
The checks that you added are only in `runWithConfig` which is only used via the command line.

Reply:
Oh, I see.  Yes, I did fail to gather that.

I think I am inclined to let purely internal users (potentially) shoot themselves in the foot arbitrarily, but it's not a strong preference so I would be fine with going the other way if you'd rather.